test/extensions/clusters/aggregate/cluster_integration_test.cc

```diff
@@ -178,6 +211,50 @@ class AggregateIntegrationTest
     xds_stream_->startGrpcStream();
   }
 
+  void setCircuitBreakerLimits(envoy::config::cluster::v3::Cluster& cluster,
+                               const CircuitBreakerLimits& limits) {
+    auto* cluster_circuit_breakers = cluster.mutable_circuit_breakers();
+
+    auto* cluster_circuit_breakers_threshold_default = cluster_circuit_breakers->add_thresholds();
+    cluster_circuit_breakers_threshold_default->set_priority(
+        envoy::config::core::v3::RoutingPriority::DEFAULT);
+
+    cluster_circuit_breakers_threshold_default->mutable_max_connections()->set_value(
+        limits.max_connections);
+    cluster_circuit_breakers_threshold_default->mutable_max_pending_requests()->set_value(
+        limits.max_pending_requests);
+    cluster_circuit_breakers_threshold_default->mutable_max_requests()->set_value(
+        limits.max_requests);
+    cluster_circuit_breakers_threshold_default->mutable_max_retries()->set_value(
+        limits.max_retries);
+    cluster_circuit_breakers_threshold_default->mutable_max_connection_pools()->set_value(
+        limits.max_connection_pools);
+    cluster_circuit_breakers_threshold_default->set_track_remaining(true);
+  }
+
+  void setMaxConcurrentStreams(envoy::config::cluster::v3::Cluster& cluster,
+                               uint32_t max_concurrent_streams) {
+    envoy::extensions::upstreams::http::v3::HttpProtocolOptions http_protocol_options;
+    http_protocol_options.mutable_explicit_http_config()
+        ->mutable_http2_protocol_options()
+        ->mutable_max_concurrent_streams()
+        ->set_value(max_concurrent_streams);
+    (*cluster.mutable_typed_extension_protocol_options())
+        ["envoy.extensions.upstreams.http.v3.HttpProtocolOptions"]
+            .PackFrom(http_protocol_options);
+  }
+
+  void
+  reduceAggregateClustersListToOneCluster(envoy::config::cluster::v3::Cluster& aggregate_cluster) {
+    auto* aggregate_cluster_type = aggregate_cluster.mutable_cluster_type();
+    auto* aggregate_cluster_typed_config = aggregate_cluster_type->mutable_typed_config();
+    envoy::extensions::clusters::aggregate::v3::ClusterConfig new_aggregate_cluster_typed_config;
+    aggregate_cluster_typed_config->UnpackTo(&new_aggregate_cluster_typed_config);
+    new_aggregate_cluster_typed_config.clear_clusters();
+    new_aggregate_cluster_typed_config.add_clusters("cluster_1");
+    aggregate_cluster_typed_config->PackFrom(new_aggregate_cluster_typed_config);
+  }
```

Comment:
These call all be free functions in an anonymous namespace since they don't rely on any of the members of the test fixture.